lib/RestructureCFG/InlineDispatcherSwitch.cpp

```diff
@@ -509,6 +509,65 @@ static void processNestedWeavedSwitches(SwitchNode *Switch) {
   }
 }
 
+static RecursiveCoroutine<bool> containsNonLocalCF(ASTNode *Node) {
+  switch (Node->getKind()) {
+  case ASTNode::NK_List: {
+    SequenceNode *Seq = llvm::cast<SequenceNode>(Node);
+
+    bool HasNonLocalCF = false;
+    for (ASTNode *N : Seq->nodes()) {
+      HasNonLocalCF = HasNonLocalCF or rc_recur containsNonLocalCF(N);
+    }
```

Comment:
Wouldn't it be faster to iterate in reverse order?
The very moment `HasNonLocalcCF` is set to true, you can just bail out.

Reply:
Right, changed this [here](https://github.com/revng/revng-c/blob/f15b7b1a9506a37ed6d348f7dd6d880b1d93df16/lib/RestructureCFG/InlineDispatcherSwitch.cpp#L519)